src/sentry/api/endpoints/organization_dashboard_details.py

```diff
@@ -217,10 +217,31 @@ def put(self, request: Request, organization, dashboard) -> Response:
         if not features.has(EDIT_FEATURE, organization, actor=request.user):
             return Response(status=404)
 
+        if not request.user.is_authenticated:
+            return Response(status=401)
```

Comment:
Had to add this because the calls to the helpers from the `DashboardFavoriteUser` object managers were complaining. `user_id` is typed as an int and without this call, the typing of the `request.user` object is possibly `AnonymousUser` which I suppose doesn't have an ID

Checking for auth fixes the type issue